message_adapter/cumulus_message.py

```diff
@@ -0,0 +1,263 @@
+        matches = re.findall(template_regex, json_path_string)
+        for match in matches:
+            match_data = parse(match.lstrip('{').rstrip('}')).find(event)
+            if match_data:
+                json_path_string = json_path_string.replace(match, match_data[0].value)
+        return json_path_string
+
+    return json_path_string
+
+
+def resolve_input(event, config):
+    """
+    * Given a Cumulus message and its config, returns the input object to send to the
+    * task, as defined under config.cumulus_message
+    * @param {*} event The Cumulus message
+    * @param {*} config The config object
+    * @returns {*} The object to place on the input key of the task's event
+    """
+    if ('cumulus_message' in config and 'input' in config['cumulus_message']):
+        input_path = config['cumulus_message']['input']
+        return resolve_path_str(event, input_path)
+    return event.get('payload')
+
+
+def resolve_config_templates(event, config):
+    """
+    * Given a config object containing possible JSONPath-templated values, resolves
+    * all the values in the object using JSONPaths into the provided event.
+    *
+    * @param {*} event The event that paths resolve against
+    * @param {*} config A config object, containing paths
+    * @returns {*} A config object with all JSONPaths resolved
+    """
+    task_config = config.copy()
+    if 'cumulus_message' in task_config:
+        del task_config['cumulus_message']
+    return _resolve_config_object(event, task_config)
+
+
+def store_remote_response(incoming_event, max_size, config_keys):
+    """
+    * Stores part of a response message in S3 if it is too big to send to StepFunctions
+    * @param {*} incoming_event  - The response message
+    * @param {*} max_size        - The maximum size (in bytes) a response message portion
+    *                              can be before the method will store it in s3
+    * @param {*} config_keys     - A list of valid CMA configuration keys
+    * @returns {*} A response message, possibly referencing an S3 object for its contents
+    """
+    event = deepcopy(incoming_event)
+    replace_config = event.get('ReplaceConfig', None)
+    if not replace_config:
+        return event
+    # Set default value if FullMessage flag set
+    if replace_config.get('FullMessage', False):
+        replace_config['Path'] = '$'
+
+    replace_config_values = _parse_remote_config_from_event(replace_config, max_size)
+
+    for key in config_keys:
+        if event.get(key):
+            del event[key]
+    cumulus_meta = deepcopy(event['cumulus_meta'])
+    replacement_data = replace_config_values['parsed_json_path'].find(event)
+    if len(replacement_data) != 1:
+        raise Exception(f'JSON path invalid: {replace_config_values["parsed_json_path"]}')
+    replacement_data = replacement_data[0]
+
+    estimated_data_size = len(json.dumps(replacement_data.value))
+    if sys.version_info.major > 3:
```

Comment:
Do we still need this check if we are Py3 exclusive?

Reply:
Ty, good catch. 316c413